### PR TITLE
New Formula: muParserX

### PR DESCRIPTION
**Pull request by leezii**
A C++ math parser library with array and string support.

### CODE REVIEW

File: Library/Formula/muparserx.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+class Muparserx < Formula
+  homepage "http://muparser.beltoforion.de/"
+
+  stable do
```
Comment (mistydemeo) [inline Library/Formula/muparserx.rb]: These don't need to go inside a `stable` block unless a stable-only patch or resource accompanies them.

File: Library/Formula/muparserx.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    url "https://muparserx.googlecode.com/files/muparserx_v2_1_6.zip"
+    sha1 "f5b6ed13594c14f408168f6c0111ba4078e4b12f"
+  end
+  head "http://muparserx.googlecode.com/svn/trunk/"
+
+  depends_on "cmake" => :build
+
+  resource "CMakeLists.txt" do
+    url "http://leezii.com/CMakeLists.txt"
```
Comment (mistydemeo) [inline Library/Formula/muparserx.rb]: This isn't versioned? It doesn't come with the software?

Comment (mistydemeo) [inline Library/Formula/muparserx.rb]: It also looks like this isn't used?

Comment (leezii) [inline Library/Formula/muparserx.rb]: @mistydemeo When I put version "2.1.6", I got one audit problem
* Stable: version 2.1.6 is redundant with version scanned from URL

What do you mean "It also looks like this isn't used?"

Comment (mikemcquaid) [inline Library/Formula/muparserx.rb]: > What do you mean "It also looks like this isn't used?"

You don't use this `resource` anywhere.

File: Library/Formula/muparserx.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+  def install
+    (buildpath/"CMakeLists.txt").write <<-EOS.undent
+      cmake_minimum_required(VERSION 2.8)
+      aux_source_directory(parser SRC)
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+      add_library(Parser-dynamic SHARED ${SRC})
+      add_library(Parser-static ${SRC})
+      set_target_properties(Parser-dynamic Parser-static PROPERTIES OUTPUT_NAME muparserx ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib" LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib")
+    EOS
```
Comment (mikemcquaid) [inline Library/Formula/muparserx.rb]: Is there no buildsystem provided with the project?

File: Library/Formula/muparserx.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  test do
+    (testpath/"main.cpp").write <<-EOS.undent
+      #include <muparserx/mpParser.h>
+      int main()
+      {
+         mup::ParserX parser;
+         return 0;
+      }
+    EOS
+    system ENV.cxx, "-std=c++11", "-I#{HOMEBREW_PREFIX}/include", "-L#{HOMEBREW_PREFIX}/lib", "-lmuparserx", "main.cpp", "-o", "test"
```
Comment (mikemcquaid) [inline Library/Formula/muparserx.rb]: Shouldn't need the `HOMEBREW_PREFIX` lines; use `#{include}` and `#{lib}` instead.

File: Library/Formula/muparserx.rb (hunk 3 above)
Comment (leezii) [inline Library/Formula/muparserx.rb]: It provides one makefile, but it uses "llvm-config". I think this is not necessary. Using cmake is simple.

Comment (mikemcquaid) [inline Library/Formula/muparserx.rb]: Let's use their `Makefile` instead, please. Alternatively, submit this CMake buildsystem upstream and we'll use it when it's been officially accepted.

**Comment by jacknagel** [comment]
ping, can we get the comments addressed?

**Pull request closed without merging**